include/ensmallen_bits/nsga2/nsga2_impl.hpp

```diff
@@ -139,9 +142,9 @@ typename MatType::elem_type NSGA2::Optimize(
     BinaryTournamentSelection(population, lowerBound, upperBound);
 
     // Evaluate the objectives for the new population.
-    calculatedObjectives.resize(population.size());
-    for (size_t i = 0; i < population.size(); i++)
-      calculatedObjectives[i] = arma::Col<ElemType>(numObjectives, arma::fill::zeros);
+    calculatedObjectives.resize(population.size()); //TODO: Costly? Doing because population may change??
+    std::fill(calculatedObjectives.begin(), calculatedObjectives.end(),
```

Comment:
I'd be wary of replacing simple loops with algorithms from the std namespace. I don't think there is any performance advantage -- in the end the code will do exactly the same thing.  At the same time, this change makes things harder to read/understand, which adds to the technical debt (ie. increases the maintenance burden).

Reply:
I had a feeling I was overdoing it. Thanks for the review.